modules/pre-commit.nix

```diff
@@ -320,41 +288,19 @@ in {
             # This happens in pure shells, including lorri
             echo 1>&2 "WARNING: nix-pre-commit-hooks: git command not found; skipping installation."
           else
-            local doInstall=false
-
             # These update procedures compare before they write, to avoid
             # filesystem churn. This improves performance with watch tools like lorri
             # and prevents installation loops by via lorri.
 
-            if readlink .pre-commit-hooks >/dev/null \
-              && [[ $(readlink .pre-commit-hooks) == ${hooks} ]]; then
+            if readlink .pre-commit-config.yaml >/dev/null \
+              && [[ $(readlink .pre-commit-config.yaml) == ${configFile} ]]; then
               echo 1>&2 "nix-pre-commit-hooks: hooks up to date"
             else
               echo 1>&2 "nix-pre-commit-hooks: updating $PWD repo"
 
-              [ -L .pre-commit-hooks ] && unlink .pre-commit-hooks
-              ln -s ${hooks} .pre-commit-hooks
-
-              doInstall=true
-            fi
-
-            if cmp ${configFile} .pre-commit-config.yaml &>/dev/null; then
-              echo 1>&2 "nix-pre-commit-hooks: config up to date"
-            else
-              echo 1>&2 "nix-pre-commit-hooks: updating $PWD config"
-
-              # This can't be a symlink because its path is not constant,
-              # thus can not be committed and is invisible to pre-commit.
-              rm -f .pre-commit-config.yaml
-              cat ${configFile} >.pre-commit-config.yaml
-
-              doInstall=true
-            fi
+              [ -L .pre-commit-config.yaml ] && unlink .pre-commit-config.yaml
```

Comment:
It's better to call `rm` here for transition, as before this commit the file won't be a symlink.

Reply:
```suggestion
              [ -f .pre-commit-config.yaml ] && rm .pre-commit-config.yaml
```

Reply:
`-f` + `unlink` works just fine :)

But the thing is I don't think we should remove user config, especially because the user doesn't consciously invoke this in most cases.
See the new commit for a hopefully helpful error message.